prog/dftb+/lib_dftb/populations.F90

```diff
@@ -451,5 +451,64 @@ subroutine getChargePerShell(qOrb, orb, species, chargePerShell, qRef)
   end subroutine getChargePerShell
 
 
+  !> Block mulliken analysis with dense lower triangle matrices.
+  subroutine denseBlockMulliken(rhoSqr, overSqr, iSquare, qq)
+
+    !> Square (lower triangular) spin polarized density matrix
+    real(dp), intent(in) :: rhoSqr(:,:,:)
+
+    !> Square (lower triangular) overlap matrix
+    real(dp), intent(in) :: overSqr(:,:)
+
+    !> Atom positions in the row/column of square matrices
+    integer, intent(in) :: iSquare(:)
+
+    !> Mulliken block charges on output (mOrb, mOrb, nAtom, nSpin)
+    real(dp), intent(out) :: qq(:,:,:,:)
+
+    real(dp), allocatable :: tmpS(:,:)
+    real(dp), allocatable :: tmpD(:,:)
+
+    integer :: nAOs, nat, nSpin, ii, jj, kk, iat, iS, nOrb
+
+    nSpin = size(rhoSqr, dim=3)
+    nat = size(iSquare, dim=1) - 1
+    nAOs = size(rhoSqr, dim=1)
+
+    allocate(tmpS(nAOs,nAOs))
+    allocate(tmpD(nAOs,nAOs))
+
+    ! ... symmetrize overlap
+    tmpS(:,:) = overSqr + transpose(overSqr)
+    do ii = 1, nAOs
+      tmpS(ii,ii) = overSqr(ii,ii)
+    end do
+
+    qq(:,:,:,:) = 0.0_dp
+
+    do iS = 1, nSpin
+
+      ! ... symmetrize density matrix for spin channel
+      tmpD(:,:) = rhoSqr(:,:,iS) + transpose(rhoSqr(:,:,iS))
+      do ii = 1, nAOs
+        tmpD(ii,ii) = rhoSqr(ii,ii,iS)
+      end do
+
+      do iat = 1, nat
+        ii = iSquare(iat)
+        jj = iSquare(iat+1)
+        nOrb = jj - ii
+        qq(:nOrb,:nOrb,iAt,iS) = matmul(tmpS(ii:jj-1,:), tmpD(:,ii:jj-1))
+        qq(:nOrb,:nOrb,iAt,iS) = 0.5_dp * (qq(:nOrb,:nOrb,iAt,iS)&
+            & + transpose(qq(:nOrb,:nOrb,iAt,iS)))
+      end do
+
+    end do
+
+    deallocate(tmpS)
+    deallocate(tmpD)
```

Comment:
Remove deallocations (allocatables are automatically deallocated if going out of scope)